src/extensions/operations/remove-expired-adoptions-cron-handler/test/api.test.ts

```diff
@@ -0,0 +1,188 @@
+import { expect } from 'chai';
+import relativeDayUtc from 'relative-day-utc';
+import * as sinon from 'sinon';
+import operationApi from '../src/api.js';
+
+describe('Remove expired adoptions CRON handler', () => {
+	const database = {};
+	const getSchema = () => Promise.resolve({});
+	const env = {
+		DASHBOARD_URL: 'http://localhost:13010',
+	};
+
+	const itemsReadByQuery = sinon.stub();
+	const notificationsReadByQuery = sinon.stub();
+	const deleteByQuery = sinon.stub();
+	const createOne = sinon.stub();
+	const services = {
+		ItemsService: sinon.stub().returns({ readByQuery: itemsReadByQuery, deleteByQuery }),
+		NotificationsService: sinon.stub().returns({ createOne, readByQuery: notificationsReadByQuery }),
+	};
+	const context = { database, env, getSchema, services } as any;
+	let sandbox: sinon.SinonSandbox;
+
+	beforeEach(() => {
+		sinon.resetHistory();
+		sandbox = sinon.createSandbox({ useFakeTimers: { now: new Date('2023-04-25') } });
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	it('should notify user if probe is offline for >2 days', async () => {
+		itemsReadByQuery.onFirstCall().resolves([{
+			id: 'probeId1',
+			ip: '1.1.1.1',
+			userId: 'userId1',
+			status: 'offline',
+			lastSyncDate: relativeDayUtc(-2).toISOString().split('T')[0],
+		}]);
+
+		itemsReadByQuery.onSecondCall().resolves([]);
+
+		const result = await operationApi.handler({}, context);
+
+		expect(createOne.callCount).to.equal(1);
+
+		expect(createOne.args[0]).to.deep.equal([
+			{
+				recipient: 'userId1',
+				subject: 'Your probe went offline',
+				message: 'Your [probe](http://localhost:13010/probes/probeId1) with IP address **1.1.1.1** has been offline for more than 24 hours. If it does not come back online before **May 23** it will be removed from your account.',
```

Comment:
Cool idea with the link but it could have been just a relative URL without the domain 😄

Reply:
Indeed 👍